KAFKA-5134 Replace zkClient.getChildren method with zkUtils.getChildren

baluchicken opened the pull request:
@ijuma plz review. I also created a version where I didn't refactor the  getPartitionAssignmentForTopics and the getConsumersPerTopic methods, but I thought maybe it is a good idea to do and move the logic (where there is no ZK connection) to the ZkUtils Object.
About the two new methods, I wanted to make them private and visible only for testing, but because the tests where I use them are not in the same package I couldn't do it.

baluchicken commented:
Resolve conflicts.

baluchicken commented:
Resolved the conflicts

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5882/
Test PASSed (JDK 8 and Scala 2.12).

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5897/
Test FAILed (JDK 7 and Scala 2.11).

baluchicken commented:
@ijuma if you have time can you please take a look?

ijuma commented:
Can we move the changes to `getPartitionAssignmentForTopics` and `getConsumersPerTopic` to a separate PR?

ijuma commented:
To be honest, I am not clear of the benefit of moving some of the functionality to the companion object. It may not be worth it.

baluchicken commented:
@ijuma I moved the functionality to the companion object because of https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/consumer/PartitionAssignorTest.scala. In this test we are mocking zkClient. I rewrote that to mock the ZkUtil instead, but I don't want to rewrite the whole functionality of getPartitionAssignmentForTopics and getConsumersPerTopic methods, so I decided to refactor those methods move the part which does not connect to ZK to the companion object so I can reuse them during testing. But I saw that this PartitionAssignorTest.scala class is deprecated and will be removed, so I can leave this class as is and refactor every other class which uses zkClient.getChildren method. On the other hand I can reimplement the functionality which was moved to ZkUtils companion object inside this test class.

ijuma commented:
Yes, maybe we can leave deprecated classes as is if they create additional changes that are not necessary for non-deprecated classes.

baluchicken committed:
KAFKA-5134 Replace zkClient.getChildren method with zkUtils.getChildren

baluchicken commented:
Updated the PR, left the deprecated PartitionAssignorTest as is. What is the policy for removing deprecated classes?

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6190/
Test PASSed (JDK 7 and Scala 2.11).

asfgit commented:
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6174/
Test PASSed (JDK 8 and Scala 2.12).

ijuma commented:
Deprecated classes can only be removed in major releases. For major components like the old clients, they won't be removed for at least 1 year after deprecation (in addition).

ijuma reviewed (approved):
LGTM, thanks. Merging to trunk.
